File: backend/VocabuAI.Application/Learning/Generation/GenerateTextRequestDto.cs
using VocabuAI.Application.Learning.Generation.Contracts;

namespace VocabuAI.Application.Learning.Generation;

// Keep in sync with frontend/src/domain/dtos/GenerateTextRequestDto.ts

public sealed record GenerateTextRequestDto(
    Language TargetLanguage,
    int MinWordCount,
    int MaxWordCount,
    IReadOnlyCollection<GrammarConceptId> AllowedGrammar,
    IReadOnlyCollection<GrammarConceptId> ForbiddenGrammar,
    TextStyle Style,
    LanguageLevel LanguageLevel
);

File: frontend/src/domain/dtos/GenerateTextRequestDto.ts
// Keep in sync with backend/VocabuAI.Application/Learning/Generation/GenerateTextRequestDto.cs
import { LanguageLevel } from "../LanguageLevel";
import { GrammarConceptId } from "../GrammarConceptId";
import { Language } from "../Language";
import { TextStyle } from "../TextStyle";
import type { AiProvider } from "../AiProvider";

export type GenerateTextRequestDto = {
  targetLanguage: Language;
  minWordCount: number;
  maxWordCount: number;
  allowedGrammar: GrammarConceptId[];
  forbiddenGrammar: GrammarConceptId[];
  style: TextStyle;
  languageLevel: LanguageLevel;
  provider?: AiProvider | null;
};
